lib/active_model/serializer/field.rb

```diff
@@ -27,16 +27,29 @@ def value(serializer)
       def excluded?(serializer)
         case condition_type
         when :if
-          !serializer.public_send(condition)
+          !evaluate_condition(serializer)
         when :unless
-          serializer.public_send(condition)
+          evaluate_condition(serializer)
         else
           false
         end
       end
 
       private
 
+      def evaluate_condition(serializer)
+        case condition
+        when Symbol
+          serializer.public_send(condition)
+        when String
+          serializer.instance_eval(condition)
+        when Proc
+          condition.call(serializer)
+        else
+          fail ArgumentError
```

Comment:
:+1: 

By the way, it is better if raise an error on evaluate serializer's DSL I think.

https://github.com/rails-api/active_model_serializers/pull/1699/files#diff-689b80d81fb899042cc36668903c9a3dR10